examples_gbs/run_tutorial_max_clique.py

```diff
@@ -0,0 +1,155 @@
+"""
+Max Clique Tutorial
+=========================
+In this tutorial, we'll explore how to combine GBS samples with local search algorithms to
+find large cliques in graphs. Let's get started!
+
+A clique is a special type of subgraph where all possible connections between nodes are present;
+they are densest possible subgraphs of their size. The maximum clique problem, or max clique for
+short, asks the question: given a graph :math:`G`, what is the largest clique in the graph?
+Max clique is `NP-Hard <https://en.wikipedia.org/wiki/NP-hardness>`_, so finding the biggest clique
+becomes challenging for graphs with many
+nodes. This is why we need clever algorithms to identify large cliques!
+
+To get started, we'll analyze the 24-node TACE-AS graph used in :cite:`banchi2019molecular`. This
+is the *binding interaction graph* representing the spatial compatibility of atom pairs in a
+protein-molecule complex. Cliques in this graph correspond to stable docking configurations, which
+are of interest in determining how the molecule interacts with the protein.
+
+The first step is to import the Strawberry Fields GBS module and external dependencies:
+"""
+from strawberryfields.gbs import data, plot, sample, clique
+import numpy as np
+import networkx as nx
+import plotly
+
+##############################################################################
+# The adjacency matrix of the TACE-AS graph can be loaded from the ``data`` module and the
+# graph can be visualized using the :mod:`~gbs.plot` module:
+
+TA = data.TaceAs()
+A = TA.adj
+TA_graph = nx.Graph(A)
+graph_fig = plot.plot_graph(TA_graph)
+plotly.offline.plot(graph_fig, filename="TACE-AS.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/TACE-AS.html
+
+##############################################################################
+# Can you spot any cliques in the graph? It's not so easy using only your eyes! The TACE-AS graph
+# is sufficiently small that all cliques can be found by performing an exhaustive search over
+# all subgraphs. For example, below we highlight a small *maximal* clique, i.e., a clique
+# not contained inside another clique:
+
+maximal_clique = [4, 11, 12, 18]
+maximal_fig = plot.plot_graph(TA_graph, maximal_clique)
+plotly.offline.plot(maximal_fig, filename="maximal_clique.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/maximal_clique.html
+
+##############################################################################
+# We'll now use the :mod:`~clique` module to find larger cliques in the graph. We can make
+# use of the pre-generated samples from the TACE-AS graph in the ``data`` module and post-select samples with a specific number of clicks. For
+# this tutorial, we'll look at samples with eight clicks, of which there are a total of 1,984:

```

Comment:
![image](https://user-images.githubusercontent.com/49409390/66776959-a9557a80-ee95-11e9-8d22-dceb30c3ed72.png)

🤔 I think something changed when we switched to the new `postselect` function.

Reply:
Ah I got it, `postselect` works on raw samples and not on subgraph samples, and hence should be run before `to_subgraphs`.

This is something we could think about changing though.

Reply:
![image](https://user-images.githubusercontent.com/49409390/66862130-4b409a00-ef5e-11e9-9753-6f5231549929.png)

Back in business 💪